packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -402,10 +403,11 @@ export const toggleCart = () =>
         }
 
         // otherwise open the cart and load its contents
-        await Promise.all([
-            dispatch(toggleDrawer('cart')),
-            dispatch(getCartDetails())
-        ]);
+        cartActions = [dispatch(toggleDrawer('cart'))];
+        if (cart.details !== undefined && cart.details.items_count !== 0) {
+            cartActions = [dispatch(getCartDetails())];
+        }
+        await Promise.all(cartActions);
```

Comment:
The previous code,

```
await Promise.all([
  dispatch(toggleDrawer('cart')),
  dispatch(getCartDetails())
]);
```

waited for both dispatch promises to resolve (or one of them to reject) before finishing execution and letting the thunk return.

The new code doesn't replicate that behavior, even when there is something in the cart. The following factors combine to make that the case:

**`cartActions` is reassigned**

We're trying to build `cartActions` up as an array of actions to take. By the time we get to the `Promise.all` (line 410), it should have one action (`toggleDrawer`) or two (`toggleDrawer` and `getCartDetails`), depending on the number of items in the cart.

Unfortunately, line 408 is _reassigning_ `cartActions` instead of adding to it, so the `Promise.all` is only ever passed one promise, never two.

That would be tricky to debug because

**`dispatch(toggleDrawer('cart'))` is executed immediately**

The statement `cartActions = [dispatch(toggleDrawer('cart'))];` (line 406) has the effect of `toggleDrawer` executing immediately. 

Combined with the previous bullet point, this results in this thunk **not** waiting for the `toggleDrawer` dispatch to complete, which doesn't match the previous implementation and likely wasn't an intended change (correct me if I'm wrong).

Currently this action doesn't take any time at all so we don't even notice it but imagine code gets added in the future that makes it take a few seconds to complete. The thunk would return and then _sometime later_ the `toggleDrawer` dispatch would complete. Instead, we don't want to return from this thunk until that action completes.

Reply:
Note: It's fine that `dispatch(toggleDrawer('cart'))` is executed immediately, we just need to make sure we're passing that promise to `Promise.all`.

Reply:
Okay,

For this, I've done some minor modification in the code, after modification I think the behaviour is replicated with previous code which you have mentioned in this.

Please see, I've tried to print both older and newer dispatch in the developer console.
![2019_05_02_Selection_001](https://user-images.githubusercontent.com/38535982/57061904-418ee700-6cdc-11e9-8255-f147b7b17b43.png)

And In developer Console, It will return the same output.
![2019_05_02_Selection_002](https://user-images.githubusercontent.com/38535982/57061928-54a1b700-6cdc-11e9-9476-a0c480e49df6.png)

Please let me know your suggestion.

Thanks!

Reply:
Yep, you got it - thanks!